oauthproxy.go

```diff
@@ -1149,6 +1107,22 @@ func extractAllowedGroups(req *http.Request) map[string]struct{} {
 	return groups
 }
 
+// encodedState builds the OAuth state param out of our nonce and
+// original application redirect
+func encodeState(csrf cookies.CSRF, redirect string) string {
+	return fmt.Sprintf("%v:%v", csrf.HashOAuthState(), redirect)
```

Comment:
@JoelSpeed Based on the question in Slack earlier today, do we need to URLEncode `redirect` in the state?

Someone brought up an issue where they were losing secondary querystring params in the original URL -- I wonder if the `&` wasn't encoded and got passed to the IdP as a param of the Login URL?

(This was a lift and shift of old code -- but seeing it in this diff is ringing bells)

Reply:
We've been loosing those params forever, perhaps it could be because we are not URL encoding properly, it would make sense. Do you have a way you could test it easily?

Reply:
Actually - I can't recreate. When I spin up our local environment, I see multiple querystring args in the original URL getting encoded in the `state` param of the LoginURL just fine.

This make sense, since under the hood `makeLoginURL` builds them as a `url.Values` and calls `Encode()` to pop into the `url.RawQuery`

Reply:
I'm gonna drop this endeavor in this PR and handle this in another if the reporter can figure out a way to recreate the bug.